pkg/server/server.go

```diff
@@ -125,6 +129,18 @@ func (s *Server) Run(ctx context.Context) error {
 			KeyFile:       cfg.KeyFile,
 			TrustedCAFile: cfg.TrustedCAFile,
 		}
+
+		injector := make(chan *rest.Config)
+		clientLoader, err := sharding.New(s.cfg.ShardKubeconfigFile, injector)
```

Comment:
How does it behave when `shard-kubeconfig-file` is empty ? Are we compatible with the default, non-sharding behavior ?